SimPEG/Maps.py

```diff
@@ -490,6 +499,90 @@ def deriv(self, m, v=None):
     def inverse(self, m):
         return np.exp(Utils.mkvc(m))
 
+class Weighting(IdentityMap):
+    """
+        Model weight parameters.
+    """
+
+    def __init__(self, mesh=None, nP=None, weights=None, **kwargs):
+
+        if 'nC' in kwargs:
+            raise AttributeError(
+                '`nC` is depreciated. Use `nP` to set the number of model '
+                'parameters'
+            )
+
+        super(Weighting, self).__init__(mesh=mesh, nP=nP, **kwargs)
+
+        if weights is None:
+            weights = np.ones(self.nP)
+
+        self.weights = np.array(weights, dtype=float)
+        self.P = Utils.sdiag(self.weights)
```

Comment:
I wonder if we need to create this even? or maybe lazily?

because we could just multiply `m` directly.